controller/k8s/api.go

```diff
@@ -1099,3 +1105,27 @@ func isPendingOrRunning(pod *corev1.Pod) bool {
 func isFailed(pod *corev1.Pod) bool {
 	return pod.Status.Phase == corev1.PodFailed
 }
+
+func isValidRSParent(rs *appsv1.ReplicaSet) bool {
+	if rs == nil || len(rs.GetOwnerReferences()) != 1 {
+		return false
+	}
+
+	validParentKinds := []string{
+		k8s.Job,
+		k8s.ReplicationController,
```

Comment:
Is it possible for a RS to have an RC parent?

Reply:
I don't think so, my understanding is that they're similar in scope and responsibilities. I'll remove both the Cron and RC types.

Reply:
Done, added a test and removed the RC & Cron types.